src/components/App.js

```diff
@@ -38,54 +38,30 @@ const App = React.createClass({
     return { shortcuts };
   },
 
-  renderWelcomeBox() {
-    return dom.div(
-      { className: "welcomebox" },
-      L10N.getFormatStr("welcome.search", formatKeyShortcut("CmdOrCtrl+P"))
-    );
+  componentDidMount() {
+    if (isEnabled("verticalLayout")) {
+      verticalLayoutBreakpoint.addListener(this.onLayoutChange);
+    }
   },
 
-  renderCenterPane() {
-    return dom.div(
-      { className: "center-pane" },
-      dom.div(
-        { className: "editor-container" },
-        SourceTabs({
-          togglePane: this.togglePane,
-          startPanelCollapsed: this.state.startPanelCollapsed,
-          endPanelCollapsed: this.state.endPanelCollapsed,
-        }),
-        Editor(),
-        !this.props.selectedSource ? this.renderWelcomeBox() : null,
-        SourceSearch()
-      )
-    );
+  componentWillUnmount() {
+    verticalLayoutBreakpoint.removeListener(this.onLayoutChange);
   },
 
   getInitialState() {
-    const vertical = isEnabled("verticalLayout")
+    const horizontal = isEnabled("verticalLayout")
```

Comment:
I changed this because the value returned from `verticalLayoutBreakpoint` returns `false` when under 700px width which is when we go to the vertical layout, so it made sense to call it horizontal.